packages/browser-wallet/src/store/store.ts

```diff
@@ -582,11 +581,16 @@ export class Store {
         objectives.map(_.bind(this.createAndStoreRichObjectiveFromObjective, this))
       );
 
-    if (signedStates) {
-      this._eventEmitter.emit('crankRichObjectives', {
-        type: 'StatesReceived',
-        states: signedStates
-      });
+    if (signedStates?.length) {
```

Comment:
nit but I found I had to think through this check for a second.

I'd prefer a more explicit but less concise check of `signedStates && signedStates.length > 0` but this might just be a matter of preference.

Reply:
I tend to read this as: is `signedStates` defined? If so, is `signedStates.length` truthy?

Admittedly I might be a bit too enamored with optional chaining.

Reply:
I don't see optional changing as the culprit here. `signedStates.length > 0` is a more explicit clause than `signStates.length`, reducing the mental burden for the reader (at least when the reader is me).

`signedStates?.length > 0` is unfortunately not allowed. There is another, higher level change that I believe would address this and be a slight simplification for developers: why isn't `signedStates` known to be a `SignedState[]` at this point? `pushMessage2` could require `signedStates` to be defined, for instance.

When sending messages across the wire, it makes sense to make properties like `signedStates` be optional, as you save on network overhead. But at the point of deserialization, we could choose to fill in the default values `[]`. This would simplify the lives of developers (across multiple codebases), who no longer need to check if `signedStates` is defined.